src/core/types/fee_payment_detail.rs

```diff
@@ -138,16 +157,68 @@ fn generate_verifier_fee_name(verifier: &VerifierDetailV2) -> String {
         .unwrap_or_else(|| "Verifier Fee".to_string())
 }
 
+fn calc_onboarding_cost_source<S: Into<String>>(
+    verifier: &VerifierDetailV2,
+    is_retry: bool,
+    asset_type: S,
+    existing_scope_attributes: &[AssetScopeAttribute],
+) -> OnboardingCost {
+    // Always favor retry cost.  Regardless of the scenario, retries should override the specified
+    // root costs and/or subsequent classification costs
+    if is_retry {
+        return verifier.get_retry_cost();
+    }
+    let asset_type = asset_type.into();
+    // Fetch all scope attributes on the asset that used this verifier and were not for this target
+    // asset type.  If this is not empty, that means that this request is a subsequent classification
+    // for the same verifier and can use subsequent costs.
+    let other_classifications = existing_scope_attributes
+        .iter()
+        .filter(|attr| {
+            attr.verifier_address.as_str() == verifier.address && attr.asset_type != asset_type
+        })
+        .collect::<Vec<&AssetScopeAttribute>>();
+    // If at least one other classification that used this verifier is present in the existing
+    // scope attributes, then this qualifies as a subsequent classification.
+    if !other_classifications.is_empty() {
+        if let Some(ref subsequent_detail) = verifier.subsequent_classification_detail {
+            // Use the subsequent classification cost specified in the applicable asset types unless
+            // no asset types that have already been classified by this verifier are present in a
+            // provided applicable_asset_types vector.
+            if let Some(ref types) = subsequent_detail.applicable_asset_types {
```

Comment:
Kinda hate this, but dealing with `Option` isn't very fun in Rust at the moment.  Hopefully they'll smooth this syntax out over time.  Had to do some extra disgusting things around here to avoid moving values.......